x-pack/plugins/security_solution/public/common/components/item_details_card/index.tsx

```diff
@@ -0,0 +1,129 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React, { FC, isValidElement, memo, ReactElement, ReactNode, useMemo } from 'react';
+import styled from 'styled-components';
+import {
+  EuiPanel,
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiButton,
+  EuiDescriptionList,
+  EuiDescriptionListTitle,
+  EuiDescriptionListDescription,
+  EuiButtonProps,
+  PropsForButton,
+} from '@elastic/eui';
+
+const OTHER_NODES = {};
+
+const groupChildrenByType = (
+  children: ReactNode | ReactNode[],
+  types: Array<ReactElement['type']>
+) => {
+  const result = new Map<ReactElement['type'] | typeof OTHER_NODES, ReactNode[]>();
+
+  types.forEach((type) => result.set(type, []));
+  result.set(OTHER_NODES, []);
+
+  React.Children.toArray(children).forEach((child) => {
+    const key = isValidElement(child) ? child.type : OTHER_NODES;
+
+    if (!result.has(key)) {
+      result.get(OTHER_NODES)?.push(child);
+    } else {
+      result.get(key)?.push(child);
+    }
+  });
+
+  return result;
+};
+
+const SummarySection = styled(EuiFlexItem)`
+  background-color: ${({ theme }) => theme.eui.euiColorLightestShade};
+  padding: ${({ theme }) => theme.eui.euiSize};
+`;
+
+const DetailsSection = styled(EuiFlexItem)`
+  padding: ${({ theme }) => theme.eui.euiSize};
+`;
+
+const DescriptionListTitle = styled(EuiDescriptionListTitle)`
+  width: 40%;
+`;
+
+const DescriptionListDescription = styled(EuiDescriptionListDescription)`
+  width: 60%;
+`;
+
+interface ItemDetailsPropertySummaryProps {
+  name: ReactNode | ReactNode[];
+  value: ReactNode | ReactNode[];
+}
+
+export const ItemDetailsPropertySummary: FC<ItemDetailsPropertySummaryProps> = memo(
+  ({ name, value }) => (
+    <>
+      <DescriptionListTitle>{name}</DescriptionListTitle>
+      <DescriptionListDescription>{value}</DescriptionListDescription>
+    </>
+  )
+);
+
+ItemDetailsPropertySummary.displayName = 'ItemPropertySummary';
+
+export const ItemDetailsAction: FC<PropsForButton<EuiButtonProps>> = memo(
+  ({ children, ...rest }) => (
+    <EuiFlexGroup direction="column" alignItems="flexEnd" justifyContent="flexEnd">
+      <EuiFlexItem grow={false}>
+        <EuiButton {...rest}>{children}</EuiButton>
+      </EuiFlexItem>
+    </EuiFlexGroup>
+  )
+);
+
+ItemDetailsAction.displayName = 'ItemDetailsAction';
+
+export const ItemDetailsCard: FC = memo(({ children }) => {
```

Comment:
Its the first time I'm seeing this type of component, so let just see if I got this right:

this Component can take in any type of `children` allowed by React.Children type, and then it groups those components up by Type (a specific React component type) and uses those groupings to actually build the returned card.

Do I have that right?

Reply:
One other request. Because the approach taken with this component is so different from existing ones, can you add TS typings to the `children` prop so that consumers of this can have a bit of help understanding how to use it?

Reply:
yes.